ksql-serde/src/main/java/io/confluent/ksql/serde/json/KsqlJsonDeserializer.java

```diff
@@ -115,66 +137,93 @@ private static Object enforceFieldType(final JsonValueContext context) {
   }
 
   private static String processString(final JsonValueContext context) {
-    if (context.val instanceof Map) {
-      return context.deserializer.gson.toJson(context.val);
+    if (context.val instanceof ObjectNode) {
+      try {
+        // this ensure sorted order, there's an issue with Jackson where just enabling
```

Comment:
Can you explain why sorted order is necessary?

Reply:
This is necessary because some of our testing asserts the output of a serialized JSON string. Specifically, if you look at `simple-struct.json` we specify a stream:
```
CREATE STREAM orders (... itemid VARCHAR, ...
```
But when we produce to it, we produce JSON structs:
```
            "ITEMID": {
              "ITEMID": 6,
              "CATEGORY": {
                "ID": 2,
                "NAME": "Food"
              },
              "NAME": "Item_6"
            },
```
Then when we check to see that we have the right value, we check for:
```
            "ITEMID": "{\"CATEGORY\":{\"ID\":2,\"NAME\":\"Food\"},\"ITEMID\":6,\"NAME\":\"Item_6\"}",
```
Note that this is ordered alphabetically, not in the order it was produced. Something I did (probably bypassing `JsonConverter`) made us no longer sort it, so I added that back for consistency.

Reply:
I would worry about adding sorting (sorting is potentially a slow operation) in what Serde code which is potentially performance sensitive. Or putting any code in production code that's only there to satisfy test code. Is there no way to sort the output in the test before you assert it?

Reply:
Let me dig into it a little more to see where the sorting was being done, but my intention was less to satisfy the test and more to maintain backwards compatibility. If someone was depending on this behavior, we can't just remove it.

Reply:
Ok makes sense. But JSON objects have no defined order (as per spec), so if anyone is relying on that order, I'd consider that a bug in their code, unless we have specifically documented that we sort our JSON alphabetically. The reason I say that is we don't want to be shackled with the burden of having to sort it for ever, which could cause a real problem when we want to optimise our serde code to get the best performance :)

Reply:
Do we still need the sorted deserialization? It looks like you updated the tests to not assume sorted order (?)

Reply:
I did not update the tests (which I guess is confusing, because they don't show up 😂 ) - if we want to do this is should be in a separate PR (https://github.com/confluentinc/ksql/issues/3664)

Reply:
I'm with Tim here - we shouldn't be paying the cost in our production code just to satisfy the tests.  We should look to remove this with all haste!

I've tagged the above issue as 6.0, though I think that's the _latest_ we should fix it.